scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -2081,15 +2086,20 @@ class FormatOps(
     }
 
     object BlockImpl extends Impl {
+
       def tryGetSplits(ft: FormatToken, nft: FormatToken)(implicit
-          style: ScalafmtConfig
+          style: ScalafmtConfig,
+          line: sourcecode.Line
       ): Option[Seq[Split]] = {
         val leftOwner = ft.meta.leftOwner
         findTreeWithParentSimple(nft.meta.rightOwner)(_ eq leftOwner) match {
           case Some(t: Term.Block)
-              if t.stats.lengthCompare(1) > 0 && isBlockStart(t, nft) =>
+              if (t.stats.lengthCompare(1) > 0 ||
```

Comment:
you don't need to check length anymore. just remove this line and replace`exists` with `forall` on next.

Reply:
Wait but if I dontt check the length then stats with a single Term will also get caught in this case, no ?

Reply:
We want either a block with many terms or a block with a single non term. It seems to me exclusive.

Reply:
Changed it  all a bit, let me know what you think. I think we still need the compare the length.